src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,74 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+import classnames from 'classnames';
+import { createComment } from '../api/comments';
+import { PostsContext } from '../context/PostsContext';
+import { CommentsContext } from '../context/CommentsContext';
+import { Comment } from '../types/Comment';
+
+const defaultFormData = {
+  name: '',
+  email: '',
+  body: '',
+};
 
 export const NewCommentForm: React.FC = () => {
+  const { post } = useContext(PostsContext);
+  const { setComments } = useContext(CommentsContext);
+
+  const [formData, setFormData] = useState(defaultFormData);
+  const [hasErrorForm, setHasErrorForm] = useState(false);
+
+  const handleChangeForm = (
+    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    event.preventDefault();
+
+    setFormData(prev => ({ ...prev, [event.target.name]: event.target.value }));
+  };
```

Comment:
```suggestion
  const handleChangeForm = (
    event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
  ) => {
    const { name, value } = event.target;

    setFormData(prev => ({ ...prev, [name]: value }));
  };
```
Why do you do `event.preventDefault();`?